src/dune_rules/melange/melange_rules.ml

```diff
@@ -248,41 +248,85 @@ let setup_emit_cmj_rules ~sctx ~dir ~scope ~expander ~dir_contents
   let* () = Buildable_rules.gen_select_rules sctx compile_info ~dir in
   Buildable_rules.with_lib_deps ctx compile_info ~dir ~f
 
+let raise_external_runtime_dep_error ~loc lib_name path =
+  User_error.raise ~loc
+    [ Pp.textf
+        "Public library %s depends on external path `%s'. This is not allowed."
+        (Lib_name.to_string lib_name)
+        (Path.to_string path)
+    ]
+    ~hints:
+      [ Pp.textf
+          "Move the external dependency to the workspace and use a relative \
+           path."
+      ]
+
 module Runtime_deps = struct
-  let to_action_builder ~expander dep_conf =
-    let runtime_deps, _sandbox =
-      Dep_conf_eval.unnamed_get_paths ~expander dep_conf
+  let targets ~output ~for_ deps =
+    let raise_external_dep_error src =
+      let lib_info =
+        match for_ with
+        | `Library lib_info -> lib_info
+        | `Emit -> assert false
+      in
+      let loc =
+        match Lib_info.melange_runtime_deps lib_info with
+        | Local (loc, _) -> loc
+        | External _ -> Loc.none
```

Comment:
Shouldn't this be `assert  false` as well? If the runtime deps are external, the library must have been installed. Therefore, all of the paths in it cannot be external deps.

Reply:
Indeed. Fixed